include/openssl/core.h

```diff
@@ -157,10 +157,14 @@ struct ossl_param_st {
  * |in|         is the array of functions that the Core passes to the provider.
  * |out|        will be the array of base functions that the provider passes
  *              back to the Core.
+ * |provctx|    a provider side context object, optionally created if the
+ *              provider needs it.  This value is passed to other provider
+ *              functions, notably other context constructors.
  */
 typedef int (OSSL_provider_init_fn)(const OSSL_PROVIDER *provider,
```

Comment:
I'm now wondering about the value of the OSSL_PROVIDER * arg here. It kind of seems to me that its purpose has been replaced by the provider ctx.

Reply:
I think you need to look at this:

https://github.com/openssl/openssl/blob/96384e613ae7092fb6f63daa69a9601d128416b2/crypto/provider_core.c#L621-L642

Reply:
Ah. Ok.